frontend/src/lib/components/CommandPalette/index.scss

```diff
@@ -0,0 +1,96 @@
+.palette__container {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+}
+
+.palette__box {
+    z-index: 99999;
+    position: fixed;
+    top: 30%;
+    display: flex;
+    flex-direction: column;
+    width: 36rem;
+    color: #fff;
+}
+
+.palette__row {
+    display: flex;
+    align-items: center;
+    height: 4rem;
+    width: 100%;
+    padding: 0 1.875rem;
+    font-size: 1rem;
+    line-height: 4rem;
+}
+
+.palette__row--small {
+    height: 1.5rem;
+    line-height: 1.5rem;
+    font-size: 0.75rem;
+    text-transform: uppercase;
+    font-weight: bold;
+}
+
+.palette__display {
+    padding-left: 1.5rem;
+    font-size: 1rem;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    overflow: hidden;
+}
+
+.palette__input {
+    flex-grow: 1;
+    border: none;
+    outline: none;
+    background: transparent;
+    color: #fff;
+    overflow-y: scroll;
+}
+
+.palette__results-container {
+    overflow-y: scroll;
+}
+
+.palette__result {
+    height: 4rem;
+    width: 100%;
+    padding: 0 1.875rem;
+    display: flex;
+    align-items: center;
+    font-size: 1rem;
+    position: relative;
+    cursor: pointer;
+}
+
+.palette__result--focused {
+    background-color: rgba(0, 0, 0, 0.35);
+
+    &:before {
+        background-color: #1890ff;
+        content: '';
+        position: absolute;
+        left: 0;
+        top: 0;
+        bottom: 0;
+        width: 0.375rem;
+    }
+}
+
+.palette__scope {
+    background-color: rgba(255, 255, 255, 0.1);
+    color: rgba(255, 255, 255, 0.8);
+}
+
+.palette__icon {
+    display: flex;
+    align-items: center;
+    width: 1rem;
+    height: 100%;
+}
```

Comment:
Great for the BEM syntax, though since we're using SCSS, if possible it's nice to just nest elements in a very unique global class (`.command-palette { .container { .box { ... }  }  }`). This greatly reduces any kind of future conflicts... yet of course nested classes with simple names like `.palette { .icon { ... } }` will possibly have something bleed into them.